### PR TITLE
Typescript LLD groundwork

### PR DESCRIPTION
**Pull request by elbywan**
### 📝 Description

**Groundwork for the typescript migration.**

- Add types for all non-renderer parts of LLD.
- Re-enable the CI type checking for these parts.

### ❓ Context

- **Impacted projects**: `lld` 
- **Linked resource(s)**: [confluence](https://ledgerhq.atlassian.net/wiki/spaces/WALLETCO/pages/3960111327/LLD+Typescript) 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/main/window-lifecycle.ts [hunk 1]
```diff
@@ -102,7 +108,7 @@ export async function createMainWindow({ dimensions, positions }: any, settings:
   if (DEV_TOOLS && !process.env.DISABLE_DEV_TOOLS) {
     mainWindow.webContents.on("did-frame-finish-load", () => {
       if (mainWindow) {
-        mainWindow.webContents.once("devtools-open", () => {
+        mainWindow.webContents.once("devtools-opened", () => {
```
Comment (gre) [inline apps/ledger-live-desktop/src/main/window-lifecycle.ts:111]: does it mean we had a bug?

File: apps/ledger-live-desktop/src/main/window-lifecycle.ts [hunk 2]
```diff
@@ -112,6 +118,7 @@ export async function createMainWindow({ dimensions, positions }: any, settings:
   mainWindow.on("closed", () => {
     mainWindow = null;
   });
+  // @ts-expect-error Removed in Electron 22. See: https://www.electronjs.org/docs/latest/breaking-changes#removed-webcontents-new-window-event
```
Comment (gre) [inline apps/ledger-live-desktop/src/main/window-lifecycle.ts:121]: so it's possibly dead code?

**Review by gre — COMMENTED** [review]
been reviewing the code at the present stage. so far so good, some comments.

File: apps/ledger-live-desktop/src/main/window-lifecycle.ts (hunk 1 above)
Comment (elbywan) [reply apps/ledger-live-desktop/src/main/window-lifecycle.ts:111]: I believe so yes :(

https://www.electronjs.org/docs/latest/api/web-contents#event-devtools-opened

File: apps/ledger-live-desktop/src/main/window-lifecycle.ts (hunk 2 above)
Comment (elbywan) [reply apps/ledger-live-desktop/src/main/window-lifecycle.ts:121]: worse, it's a piece of code that was working with electron 22- and that has not been migrated over to electron 22+ 

I am focusing on typing so I did not touch it, but I will add a `// TODO:` comment to outline it. (I thought we were using electron 21 and did not double check 🤦)

https://www.electronjs.org/docs/latest/breaking-changes#removed-webcontents-new-window-event

File: apps/ledger-live-desktop/src/internal/live-common-setup.ts [hunk 3]
```diff
@@ -31,13 +31,15 @@ if (getEnv("DEVICE_PROXY_URL")) {
...
-    open: devicePath =>
+    open: () =>
       retry(() => TransportNodeHidSingleton.open(), {
         maxRetry: 4,
       }),
     disconnect: () => Promise.resolve(),
     setAllowAutoDisconnect: (transport, _, allow) => {
-      transport.setAllowAutoDisconnect(allow);
+      return Promise.resolve(
+        (transport as TransportNodeHidSingleton).setAllowAutoDisconnect(allow),
```
[the remark this answers is not in the document]
Comment (elbywan) [reply apps/ledger-live-desktop/src/internal/live-common-setup.ts:41]: it was to please to type system (function expects a promise as the return type), I can try with await but it should be essentially the same thing

**Review by gre — APPROVED** [review]

**Review by valpinkman — APPROVED** [review]
LETS GO

**Pull request merged**